src/types/test/fun-extends-opt-self.tact

```diff
@@ -0,0 +1,46 @@
+primitive Int;
+
+fun test(x: Int, y: Int?): Int {
+    if (y == null) {
+        return x;
+    }
+    return x + y!!;
+}
+
+extends fun test_extends(self: Int, y: Int?): Int {
+    if (y == null) {
+        return self;
+    }
+    return self + y!!;
+}
+
+extends fun test_extends_self(self: Int?, y: Int): Int {
+    if (self == null) {
+        return y;
+    }
+    return self!! + y;
+}
+
+extends mutates fun test_mutates(self: Int, y: Int?): Int {
+    if (y == null) {
+        return self;
+    }
+    self += y;
```

Comment:
`self += y` should not typecheck, but this should: `self += y!!`

since these tests do not typecheck the function bodies, let's just return some constants right away to avoid confusion